File: leetcode/435.py
from typing import List


class Solution:

    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda elem: elem[1])
        total, prev = 0, intervals[0][1]
        for i in range(1, len(intervals)):
            if prev > intervals[i][0]:
                total += 1
            else:
                prev = intervals[i][1]
        return total

File: leetcode/435.cpp
#include <vector>
#include <algorithm>

using std::sort;
using std::vector;

bool cmp(const vector<int> &a, const vector<int> &b) { return a[1] < b[1]; }

class Solution {
public:
  int eraseOverlapIntervals(vector<vector<int>> &intervals) {
    sort(
        intervals.begin(), intervals.end(),
        [](const vector<int> &a, const vector<int> &b) { return a[1] < b[1]; });
    // sort(intervals.begin(), intervals.end(), cmp);
    int n = intervals.size();
    int prev = intervals[0][1];
    int count = 0;
    for (int i = 1; i < n; ++i) {
      if (intervals[i][0] < prev) {
        ++count;
      } else {
        prev = intervals[i][1];
      }
    }
    return count;
  }
};